activerecord/lib/active_record/database_configurations/connection_url_resolver.rb

```diff
@@ -67,14 +67,16 @@ def raw_config
               database: uri.opaque
             )
           else
-            query_hash.merge(
+            connection_attrs = {
               adapter: @adapter,
               username: uri.user,
               password: uri.password,
               port: uri.port,
               database: database_from_path,
               host: uri.hostname
-            )
+            }.reject! { |key, value| value.nil? }
```

Comment:
What do you think? or using `reject! { |key, value| value.nil? }` is faster?
```suggestion
            }.compact!
```

Reply:
I like the `compact!`. Let me run some benchmarks and I'll post the changes.